cross/minio/Makefile

```diff
@@ -0,0 +1,89 @@
+PKG_NAME = minio
+PKG_VERS = 2019-05-02T19-07-09Z
+PKG_EXT = tar.gz
+PKG_DIST_PREFIX = RELEASE.$(PKG_VERS)
+PKG_DIST_NAME = $(PKG_DIST_PREFIX).$(PKG_EXT)
+PKG_DIST_FILE = $(PKG_NAME)-$(PKG_DIST_NAME)
+PKG_DIST_SITE = https://github.com/minio/minio/archive
+PKG_DIR = $(PKG_NAME)-$(PKG_DIST_PREFIX)
+
+DEPENDS = native/go
+
+HOMEPAGE = https://min.io/
+COMMENT  = The 100% Open Source, Enterprise-Grade, Amazon S3 Compatible Object Storage.
+LICENSE  = Apache License 2.0
+
+CONFIGURE_TARGET = minio_configure
+COMPILE_TARGET = minio_compile
+INSTALL_TARGET = minio_install
+
+GOOS = linux
+CGO_ENABLED = 0
+
+# set this to 1 to create static linked binaries
+STATIC_BINARIES = 0
+
+include ../../mk/spksrc.cross-cc.mk
+
+UNSUPPORTED_ARCHS = $(PPC_ARCHES)
+
+# Define GO_ARCH for go compiler
+ifeq ($(findstring $(ARCH),$(ARM5_ARCHES)),$(ARCH))
+GO_ARCH = arm
+ENV += GOARM=5
+endif
+ifeq ($(findstring $(ARCH),$(ARM7_ARCHES)),$(ARCH))
+GO_ARCH = arm
+ENV += GOARM=7
+endif
+ifeq ($(findstring $(ARCH),$(ARM8_ARCHES)),$(ARCH))
+GO_ARCH = arm64
+endif
+ifeq ($(findstring $(ARCH),$(x86_ARCHES)),$(ARCH))
+GO_ARCH = 386
+endif
+ifeq ($(findstring $(ARCH),$(x64_ARCHES)),$(ARCH))
+GO_ARCH = amd64
+endif
+# Go has no ppc32 support for qoriq https://github.com/golang/go/issues/22885
+#ifeq ($(findstring $(ARCH),$(PPC_ARCHES)),$(ARCH))
+#GO_ARCH = ppc32
+#endif
+ifeq ($(GO_ARCH),)
+$(error Unsupported ARCH $(ARCH))
+endif
+
+EXTRACT_PATH = $(WORK_DIR)/src/github.com/$(PKG_NAME)
+MINIO_DIR = $(EXTRACT_PATH)/minio
+MINIO_BIN = $(MINIO_DIR)/minio
+
+# use workdir as gopath
+ENV += GOPATH=$(WORK_DIR)/src
```

Comment:
to avoid separate download of the dependent go packages I propose to use a centralized go package folder like this:
```
# use distrib folder 'go' as GOPATH to download dependencies only once
ENV += GOPATH=$(DISTRIB_DIR)/go
```
This fastens the build `all-supported` and saves here 30GB and the respective download time.

Unfortuatly this works only for minio and syncthing packages.
The fritzctl (synocli-net) and dnscrypt-proxy fail to build with this GOPATH.

@ymartin59 may be we should document how to build packages written in go and put some code into common make files (spksrc-go.mk) for GOARCH,GOPATH etc.?

Reply:
Added in https://github.com/SynoCommunity/spksrc/pull/3689/commits/07af9ad3f04e5fb36e20dc3d2a5901ba3787c9f1 until a generic solution is available.